sdk/ai/azure-ai-client/samples/agents/async_samples/sample_agents_vector_store_batch_file_search_async.py

```diff
@@ -0,0 +1,105 @@
+    the Azure Agents service using a asynchronous client.
+
+USAGE:
+    python sample_agents_vector_store_batch_file_search_async.py
+
+    Before running the sample:
+
+    pip install azure.ai.client azure-identity
+
+    Set this environment variables with your own values:
+    AI_CLIENT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+"""
+
+import asyncio
+import os
+from azure.ai.client.aio import AzureAIClient
+from azure.ai.client.models import FileSearchToolResource, FileSearchToolDefinition, ToolResources, FilePurpose
+from azure.identity import DefaultAzureCredential
+
+
+async def main():
+    # Create an Azure AI Client from a connection string, copied from your AI Studio project.
+    # At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
+    # Customer needs to login to Azure subscription via Azure CLI and set the environment variables
+
+    ai_client = AzureAIClient.from_connection_string(
+        credential=DefaultAzureCredential(),
+        conn_str=os.environ["AI_CLIENT_CONNECTION_STRING"]
+    )
+
+    # Or, you can create the Azure AI Client by giving all required parameters directly
+    """
+    ai_client = AzureAIClient(
+        credential=DefaultAzureCredential(),
+        host_name=os.environ["AI_CLIENT_HOST_NAME"],
+        subscription_id=os.environ["AI_CLIENT_SUBSCRIPTION_ID"],
+        resource_group_name=os.environ["AI_CLIENT_RESOURCE_GROUP_NAME"],
+        workspace_name=os.environ["AI_CLIENT_WORKSPACE_NAME"],
+        logging_enable=True, # Optional. Remove this line if you don't want to show how to enable logging
+    )
+    """
+
+    async with ai_client:
+        
+        # upload a file and wait for it to be processed
+        file = await ai_client.agents.upload_file_and_poll(file_path="../product_info_1.md", purpose=FilePurpose.AGENTS)    
+        print(f"Uploaded file, file ID: {file.id}")
+
+        # create a vector store with no file and wait for it to be processed
+        # if you do not specify a vector store, create_message will create a vector store with a default expiration policy of seven days after they were last active 
+        vector_store = await ai_client.agents.create_vector_store_and_poll(file_ids=[], name="sample_vector_store")
+        print(f"Created vector store, vector store ID: {vector_store.id}")
+        
+        # add the file to the vector store
+        vector_store_file_batch = await ai_client.agents.create_vector_store_file_batch_and_poll(vector_store_id=vector_store.id, file_ids=[file.id])    
+        print(f"Created vector store file batch, vector store file batch ID: {vector_store_file_batch.id}")
+        
+        # create a tool resources object with the file search tool
+        file_search_tool_resource = FileSearchToolResource(vector_store_ids=[vector_store.id])
```

Comment:
not sure if in this sample we want to show FileSearchTool + ToolSet

Reply:
maybe not as there is no many tools

Reply:
```
    file_search_tool = FileSearchTool()
    file_search_tool.add_vector_store_list([vector_store.id])

    agent = ai_client.agents.create_agent(
        model="gpt-4-1106-preview", name="my-assistant", 
        instructions="You are helpful assistant",
        tools=file_search_tool.definitions,
        tool_resources=file_search_tool.resources
    )
```

Reply:
Thanks Jarno..   I am making

```
file_search_tool = FileSearchTool(vector_store_ids=[vector_store.id])
```